demos/http3/ossl-nghttp3-demo-server.c

```diff
@@ -15,290 +15,312 @@
-               (unsigned long long) ssl_ids[i].id);
-        hassomething++;
-      }
-    } else {
-      /* Figure out ??? */
-      printf("revent %llu (%d) on %llu\n", (unsigned long long) item->revents, SSL_POLL_EVENT_W,
-             (unsigned long long) ssl_ids[i].id);
+    if (item->revents & SSL_POLL_EVENT_OSU) {
+        /* at least one uni */
+        printf("Create uni\n");
+        /* we have 4 streams from the client 2, 6 , 10 and 0 */
+        /* need 2 streams to the client */
+        if (!h3ssl->has_uni) {
+            SSL *rstream;
+            SSL *pstream;
+            uint64_t r_streamid, p_streamid;
+
+            printf("Create uni beacause !h3ssl->has_uni\n");
+            rstream = SSL_new_stream(ssl_ids[0].s, SSL_STREAM_FLAG_UNI);
+            if (rstream != NULL) {
+                fprintf(stderr, "=> Opened on %llu\n",
+                        (unsigned long long)SSL_get_stream_id(rstream));
+                fflush(stderr);
+            } else {
+                fprintf(stderr, "=> Stream == NULL!\n");
+                fflush(stderr);
+                return -1;
+            }
+            pstream = SSL_new_stream(ssl_ids[0].s, SSL_STREAM_FLAG_UNI);
+            if (pstream != NULL) {
+                fprintf(stderr, "=> Opened on %llu\n",
+                        (unsigned long long)SSL_get_stream_id(pstream));
+                fflush(stderr);
+            } else {
+                fprintf(stderr, "=> Stream == NULL!\n");
+                fflush(stderr);
+                return -1;
+            }
+            r_streamid = SSL_get_stream_id(rstream);
+            p_streamid = SSL_get_stream_id(pstream);
+            if (nghttp3_conn_bind_qpack_streams(h3conn, p_streamid,
+                                                r_streamid)) {
+                printf("nghttp3_conn_bind_qpack_streams failed!\n");
+                return -1;
+            }
+            printf("control: NONE enc %llu dec %llu\n",
+                   (unsigned long long)p_streamid,
+                   (unsigned long long)r_streamid);
+            add_id(SSL_get_stream_id(rstream), rstream, h3ssl);
+            add_id(SSL_get_stream_id(pstream), pstream, h3ssl);
+            h3ssl->has_uni = 1;
+            if (result_count == 1) {
+                printf("read_from_ssl_ids 1 event only!\n");
+                return 0; /* one event only so we are done */
+            }
+        }
     }
-  }
-  return hassomething;
+    /* Well trying... */
+    if (numitem <= 1) {
+        return 0; /* Assume nothing for the moment */
+    }
+
+    /* Process the other stream */
+    for (i = 1; i < numitem; i++) {
+        item++;
+
+        if (item->revents & SSL_POLL_EVENT_R) {
+            /* try to read */
+            size_t l = sizeof(msg2) - 1;
+
+            if (SSL_net_read_desired(ssl_ids[i].s)) {
+                ret = SSL_read(ssl_ids[i].s, msg2, l);
+            } else {
+                continue;
+            }
```

Comment:
If you invert the condition, you can drop the `else` branch.

```c
if (!cond) {
    continue;
}

ret = ...
```

Reply:
In that case, you can drop the braces too (no need around just one statement/expression)